src/OrchardCore.Modules/OrchardCore.ContentFields/Settings/TextFieldSettingsDriver.cs

```diff
@@ -22,12 +34,35 @@ public override IDisplayResult Edit(ContentPartFieldDefinition partFieldDefiniti
 
     public override async Task<IDisplayResult> UpdateAsync(ContentPartFieldDefinition partFieldDefinition, UpdatePartFieldEditorContext context)
     {
+        var contentPartFieldSettings = partFieldDefinition.GetSettings<ContentPartFieldSettings>();
         var model = new TextFieldSettings();
 
         await context.Updater.TryUpdateModelAsync(model, Prefix);
 
+        // Convert colors with #RGB format to #RRGGBB, because the color editor will use black color all the time.
+        if (contentPartFieldSettings.Editor == TextFieldColorEditor)
+        {
+            if (!HexColorRegex().IsMatch(model.DefaultValue))
```

Comment:
Normally we should do a check `if (model.Required == true)` too here. But only if we add a checkbox on the view to allow returning a null value "or empty string".

Reply:
Ok, if we have opacity it is fine enough for now. We can take a look at adding an option to the "coloris" js component later on. Need to see if community would agree too.